mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py

```diff
@@ -1,45 +1,58 @@
 # -*- coding: utf-8 -*-
-# Copyright (C) 2010 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+# Copyright 2012 Savoir-faire Linux <http://www.savoirfairelinux.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-
-from openerp import models, api, fields, _
+from odoo import models, api, fields, exceptions, _
+from datetime import datetime, timedelta
 
 
 class MgmtsystemNonconformity(models.Model):
 
     _name = "mgmtsystem.nonconformity"
     _description = "Nonconformity"
-    _rec_name = "description"
-    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    _rec_name = "name"
+    _inherit = ['mail.thread']
     _order = "create_date desc"
+    _track = {
+        'field': {
+            'mgmtsystem_nonconformity.subtype_analysis': (
+                lambda s, c, u, o, ctx=None:
+                    o["stage_id"] == "mgmtsystem_nonconformity.stage_draft"
+            ),
+            'mgmtsystem_nonconformity.subtype_pending': (
+                lambda s, c, u, o, ctx=None:
+                    o["stage_id"] == "mgmtsystem_nonconformity.stage_pending"
+            ),
+        },
+    }
 
+    @api.model
     def _default_stage(self):
         """Return the default stage."""
-        return self.env.ref('mgmtsystem_nonconformity.stage_draft')
+        return (
```

Comment:
@lasley 
thanks for the suggestion.

ValueError is the right type, by the way dont you think that its better to avoid handling an exception if possible for the sake of faster execution?